src/v/kafka/server/connection_context.cc

```diff
@@ -262,10 +264,18 @@ connection_context::dispatch_method_once(request_header hdr, size_t size) {
               return res.dispatched
                 .then_wrapped([this,
                                f = std::move(res.response),
+                               cid = std::move(cid),
                                seq,
                                correlation,
                                self,
                                s = std::move(sres)](ss::future<> d) mutable {
+                    /*
+                     * Record the time taken between sequentially processed
+                     * portions of requests, this call closes the time recorded
+                     * quota window for the next request
+                     */
+                    self->_proto.quota_mgr().record_time_quota(cid);
```

Comment:
@dotnwat is this a faithful representation of a calculation of request processing time? I was considering the calculation might be considered finished only after the second stage of the request has finished processing.